lib/cusparse/interfaces.jl

```diff
@@ -75,74 +76,143 @@ for (taga, untaga) in tag_wrappers, (wrapa, transa, unwrapa) in op_wrappers
 
-function Base.:(-)(A::Transpose{T,<:CuSparseMatrixCSR}, B::Transpose{T,<:CuSparseMatrixCSR}) where {T}
-    C = geam(one(T), A.parent, -one(T), B.parent, 'O')
-    cscC = CuSparseMatrixCSC(C)
-    return CuSparseMatrixCSR(cscC.colPtr, cscC.rowVal, cscC.nzVal, size(cscC))
+function LinearAlgebra.:(*)(A::CuSparseMatrixCSR{T,M}, B::CuSparseMatrixCSR{T,M}) where {T,M}
+    CUSPARSE.version() < v"11.5.1" && throw(ErrorException("This operation is not supported by the current CUDA version."))
+    gemm('N', 'N', one(T), A, B, 'O')
 end
-
-function Base.:(+)(A::CuSparseMatrixCSR, B::CuSparseMatrix)
-    csrB = CuSparseMatrixCSR(B)
-    return geam(one(eltype(A)), A, one(eltype(A)), csrB, 'O')
+function LinearAlgebra.:(*)(A::CuSparseMatrixCSC{T,M}, B::CuSparseMatrixCSC{T,M}) where {T,M}
+    A2 = CuSparseMatrixCSR(A)
+    B2 = CuSparseMatrixCSR(B)
+    CuSparseMatrixCSC(gemm('N', 'N', one(T), A2, B2, 'O'))
 end
-
-function Base.:(-)(A::CuSparseMatrixCSR, B::CuSparseMatrix)
-    csrB = CuSparseMatrixCSR(B)
-    return geam(one(eltype(A)), A, -one(eltype(A)), csrB, 'O')
+function LinearAlgebra.:(*)(A::CuSparseMatrixCOO{T,M}, B::CuSparseMatrixCOO{T,M}) where {T,M}
+    A2 = CuSparseMatrixCSR(A)
+    B2 = CuSparseMatrixCSR(B)
+    CuSparseMatrixCOO(gemm('N', 'N', one(T), A2, B2, 'O'))
 end
 
-function Base.:(+)(A::CuSparseMatrix, B::CuSparseMatrixCSR)
-    csrA = CuSparseMatrixCSR(A)
-    return geam(one(eltype(A)), csrA, one(eltype(A)), B, 'O')
+function SparseArrays.droptol!(A::CuSparseMatrixCOO{T,M}, tol::Real) where {T,M}
+    mask = abs.(A.nzVal) .> tol
+    rows = A.rowInd[mask]
+    cols = A.colInd[mask]
+    datas = A.nzVal[mask]
+    B = sparse(rows, cols, datas, size(A)..., fmt = :coo)
+    copyto!(A, B)
 end
 
-function Base.:(-)(A::CuSparseMatrix, B::CuSparseMatrixCSR)
-    csrA = CuSparseMatrixCSR(A)
-    return geam(one(eltype(A)), csrA, -one(eltype(A)), B, 'O')
+for SparseMatrixType in [:CuSparseMatrixCSC, :CuSparseMatrixCSR, :CuSparseMatrixCOO]
+    @eval begin
+        if $SparseMatrixType in [CuSparseMatrixCSC, CuSparseMatrixCSR]
+
+            Base.reshape(A::$SparseMatrixType{T,M}, dims::NTuple{N,Int}) where {T,N,M} = $SparseMatrixType( reshape(CuSparseMatrixCOO(A), dims) )
+
+            function SparseArrays.droptol!(A::$SparseMatrixType{T,M}, tol::Real) where {T,M}
+                B = copy(CuSparseMatrixCOO(A))
+                droptol!(B, tol)
+                copyto!(A, $SparseMatrixType(B))
+            end
+            
+        end
+
+        function _spadjoint(A::$SparseMatrixType{T,M}) where {T,M}
+            cscA = CuSparseMatrixCSC(conj(A))
+            $SparseMatrixType( CuSparseMatrixCSR(cscA.colPtr, cscA.rowVal, cscA.nzVal, reverse(size(cscA))) )
+        end
+
+        function _sptranspose(A::$SparseMatrixType{T,M}) where {T,M}
+            cscA = CuSparseMatrixCSC(A)
+            $SparseMatrixType( CuSparseMatrixCSR(cscA.colPtr, cscA.rowVal, cscA.nzVal, reverse(size(cscA))) )
+        end
```

Comment:
For COO matrices you could just switch `ColdInd` and `RowInd` pointers for `_sptranspose`.
For `_spadjoint`, you just need to add `conj(nzVal)` too.